### PR TITLE
remove inverse from LOBPCG

### PR DESCRIPTION
**Pull request by nikitaved**
There is the explicit matrix inverse in the Rayleigh-Ritz transformation used in the LOBPCG implementation.
This PR replaces it with the system of matrix equations.

### CODE REVIEW

File: torch/_lobpcg.py [hunk 1]
```diff
@@ -563,10 +563,19 @@ def _get_rayleigh_ritz_transform(self, S):
         SBS = _utils.qform(B, S)
         d_row = SBS.diagonal(0, -2, -1) ** -0.5
         d_col = d_row.reshape(d_row.shape[0], 1)
-        R = torch.cholesky((SBS * d_row) * d_col, upper=True)
-        # TODO: could use LAPACK ?trtri as R is upper-triangular
-        Rinv = torch.inverse(R)
-        return Rinv * d_col
+        # Rt := R^T
+        Rt = torch.cholesky((SBS * d_row) * d_col, upper=False)
```
Comment (albanD) [inline torch/_lobpcg.py:567]: What is the reason for the `upper=False`? It is more efficient for the `triangular_solve` below?

**Review by albanD — COMMENTED** [review]
Looks good.

Did you take a look at how LOBPCG is tested? Do we have the proper tests already to make sure that such change does not change the behavior of the algorithm?

File: torch/_lobpcg.py (hunk 1 above)
Comment (nikitaved) [reply torch/_lobpcg.py:567]: Exactly, the we need inverse to be to the left from `d`, not to the right.

**Comment by nikitaved** [comment]
I will have a look, maybe @pearu can tell more.

**Comment by albanD** [comment]
Ho I forgot about this one :p 
Should we rebase and land it?

**Comment by albanD** [comment]
@nikitaved any update on this?

**Comment by nikitaved** [comment]
@albanD , sorry, I did miss your last comment. I will update this PR to comply with the current triangular solve interface.

**Comment by albanD** [comment]
Awesome! Thanks for the quick answer!

**Comment by nikitaved** [comment]
@albanD , done!

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**